File: July-2023/Java/LongestArithmetic SubsequenceofGivenDifference.java
class Solution {
    public int longestSubsequence(int[] arr, int difference) {
       Map<Integer,Integer> map = new HashMap<>();
        int result = 0;
        for(int i=0; i<arr.length; i++){
            int prev = arr[i]-difference;
               
                int subLen = map.getOrDefault(prev,0);
              
                result = Math.max(result, subLen+1);
            
            map.put(arr[i],subLen+1);
            
        }
        return result;
    }
}

File: July-2023/C++/LongestArithmeticSubsequenceOfGivenDifference.cxx
class Solution {
public:
    int longestSubsequence(vector<int>& arr, int difference) {
        unordered_map<int, int> mp;
        int n = arr.size();
        
        int result = 0;
        for(int i = 0; i<n; i++) {
            
            int prev            = arr[i] - difference;
            
            int lengt_till_prev = mp[prev];
            
            mp[arr[i]] = lengt_till_prev + 1;
            
            result = max(result, mp[arr[i]]);
        }
        
        return result;
    }
};
